src/common/datetime/create_duration_data.ts

```diff
@@ -10,11 +10,19 @@ export const createDurationData = (
   if (typeof duration !== "object") {
     if (typeof duration === "string" || isNaN(duration)) {
       const parts = duration?.toString().split(":") || [];
+      if (parts.length === 1) {
+        return { seconds: Number(parts[0]) };
+      }
+      if (parts.length > 3) {
+        return undefined;
+      }
+      const seconds = Number(parts[2]) || 0;
+      const seconds_whole = Math.floor(seconds);
```

Comment:
Is there some reason why, is this a disfavored style? I guess I could have done that, but it feels about the same to me. 

Note we will still have to do math on the milliseconds after split since it's not as simple as loading the string after the "." into ms. Since ".5" needs to be 500, ".0001" is 0, etc.

Reply:
It just seemed more readable to me 🤷